Linking fails with zstd dependency on GitHub Actions targetting Windows

Reduced example: https://github.com/niklasf/cargo-auditable-issue-141 (where `cargo build --release` works but `cargo auditable build --release` fails)

Workflow: https://github.com/niklasf/cargo-auditable-issue-141/blob/main/.github/workflows/demo.yml

Error: https://github.com/niklasf/cargo-auditable-issue-141/actions/runs/8866193441/job/24343338399

Log snippet:

```
2024-04-28T08:15:04.6004775Z   = note: LINK : warning LNK4044: unrecognized option '/Wl,--undefined=AUDITABLE_VERSION_INFO'; ignored
2024-04-28T08:15:04.6005683Z           libzstd_sys-2baf2e10c8662d45.rlib(5eeec2f4c1e9b3b6-zstd_v03.o) : error LNK2001: unresolved external symbol ___chkstk_ms
2024-04-28T08:15:04.6006640Z           libzstd_sys-2baf2e10c8662d45.rlib(1ddd1489cc625d97-hist.o) : error LNK2001: unresolved external symbol ___chkstk_ms
2024-04-28T08:15:04.6007577Z           libzstd_sys-2baf2e10c8662d45.rlib(5eeec2f4c1e9b3b6-zstd_v06.o) : error LNK2001: unresolved external symbol ___chkstk_ms
2024-04-28T08:15:04.6008518Z           libzstd_sys-2baf2e10c8662d45.rlib(5eeec2f4c1e9b3b6-zstd_v04.o) : error LNK2001: unresolved external symbol ___chkstk_ms
2024-04-28T08:15:04.6009431Z           libzstd_sys-2baf2e10c8662d45.rlib(5eeec2f4c1e9b3b6-zstd_v01.o) : error LNK2001: unresolved external symbol ___chkstk_ms
2024-04-28T08:15:04.6010344Z           libzstd_sys-2baf2e10c8662d45.rlib(5eeec2f4c1e9b3b6-zstd_v02.o) : error LNK2001: unresolved external symbol ___chkstk_ms
2024-04-28T08:15:04.6011312Z           libzstd_sys-2baf2e10c8662d45.rlib(1ddd1489cc625d97-zstd_compress.o) : error LNK2001: unresolved external symbol ___chkstk_ms
2024-04-28T08:15:04.6012284Z           libzstd_sys-2baf2e10c8662d45.rlib(b0401a448be314bb-zdict.o) : error LNK2001: unresolved external symbol ___chkstk_ms
2024-04-28T08:15:04.6013221Z           libzstd_sys-2baf2e10c8662d45.rlib(5eeec2f4c1e9b3b6-zstd_v05.o) : error LNK2001: unresolved external symbol ___chkstk_ms
2024-04-28T08:15:04.6014149Z           libzstd_sys-2baf2e10c8662d45.rlib(5eeec2f4c1e9b3b6-zstd_v07.o) : error LNK2001: unresolved external symbol ___chkstk_ms
2024-04-28T08:15:04.6015225Z           libzstd_sys-2baf2e10c8662d45.rlib(b0401a448be314bb-zdict.o) : error LNK2019: unresolved external symbol __mingw_vfprintf referenced in function fprintf
2024-04-28T08:15:04.6016319Z           libzstd_sys-2baf2e10c8662d45.rlib(b0401a448be314bb-fastcover.o) : error LNK2001: unresolved external symbol __mingw_vfprintf
2024-04-28T08:15:04.6017298Z           libzstd_sys-2baf2e10c8662d45.rlib(b0401a448be314bb-cover.o) : error LNK2001: unresolved external symbol __mingw_vfprintf
2024-04-28T08:15:04.6018495Z           D:\a\cargo-auditable-issue-x\cargo-auditable-issue-x\target\release\deps\cargo_auditable_issue_x-477cc37b35224d7f.exe : fatal error LNK1120: 2 unresolved externals
2024-04-28T08:15:04.6019329Z           
2024-04-28T08:15:04.6019444Z 
2024-04-28T08:15:05.1457161Z error: could not compile `cargo-auditable-issue-x` (bin "cargo-auditable-issue-x") due to 1 previous error
2024-04-28T08:15:05.1457962Z 
2024-04-28T08:15:05.1458107Z Caused by:
2024-04-28T08:15:05.1472415Z   process didn't exit successfully: `set CARGO='\\?\D:\a\_temp\msys64\mingw64\bin\cargo.exe'&& set CARGO_BIN_NAME=cargo-auditable-issue-x&& set CARGO_CRATE_NAME=cargo_auditable_issue_x&& set CARGO_MANIFEST_DIR='D:\a\cargo-auditable-issue-x\cargo-auditable-issue-x'&& set CARGO_PKG_AUTHORS=''&& set CARGO_PKG_DESCRIPTION=''&& set CARGO_PKG_HOMEPAGE=''&& set CARGO_PKG_LICENSE=''&& set CARGO_PKG_LICENSE_FILE=''&& set CARGO_PKG_NAME=cargo-auditable-issue-x&& set CARGO_PKG_README=''&& set CARGO_PKG_REPOSITORY=''&& set CARGO_PKG_RUST_VERSION=''&& set CARGO_PKG_VERSION=0.0.0&& set CARGO_PKG_VERSION_MAJOR=0&& set CARGO_PKG_VERSION_MINOR=0&& set CARGO_PKG_VERSION_PATCH=0&& set CARGO_PKG_VERSION_PRE=''&& set CARGO_PRIMARY_PACKAGE=1&& set PATH='D:\a\cargo-auditable-issue-x\cargo-auditable-issue-x\target\release\deps;C:\Users\runneradmin\.rustup\toolchains\stable-x86_64-pc-windows-msvc\bin;D:\a\_temp\msys64\mingw64\bin;D:\a\_temp\msys64\usr\local\bin;D:\a\_temp\msys64\usr\bin;D:\a\_temp\msys64\usr\bin;C:\Windows\System32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;D:\a\_temp\msys64\usr\bin\site_perl;D:\a\_temp\msys64\usr\bin\vendor_perl;D:\a\_temp\msys64\usr\bin\core_perl'&& C:\Users\runneradmin\.cargo\bin\cargo-auditable.exe rustc --crate-name cargo_auditable_issue_x --edition=2021 'src\main.rs' --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --crate-type bin --emit=dep-info,link -C opt-level=3 -C embed-bitcode=no -C metadata=477cc37b35224d7f -C extra-filename=-477cc37b35224d7f --out-dir 'D:\a\cargo-auditable-issue-x\cargo-auditable-issue-x\target\release\deps' -L 'dependency=D:\a\cargo-auditable-issue-x\cargo-auditable-issue-x\target\release\deps' --extern 'zstd=D:\a\cargo-auditable-issue-x\cargo-auditable-issue-x\target\release\deps\libzstd-c8cf5de31f378c0a.rlib' -L 'native=D:\a\cargo-auditable-issue-x\cargo-auditable-issue-x\target\release\build\zstd-sys-33d347c9f0b79841\out'` (exit code: 1)
2024-04-28T08:15:05.1585513Z ##[error]Process completed with exit code 1
```

New run: https://github.com/niklasf/cargo-auditable-issue-141/actions/runs/8866914352/job/24344898298

> unrecognized option '/Wl,--undefined=AUDITABLE_VERSION_INFO'; ignored

This is now fixed.

> However, it is only a warning, so that's not what is causing compilation to fail.

And indeed it still fails with `libzstd_sys-2baf2e10c8662d45.rlib(b0401a448be314bb-fastcover.o) : error LNK2001: unresolved external symbol __mingw_vfprintf` etc.

--- a/cargo-auditable/src/rustc_wrapper.rs
+++ b/cargo-auditable/src/rustc_wrapper.rs
@@ -52,6 +52,8 @@ pub fn main(rustc_path: &OsStr) {
                     // Prevent the symbol from being removed as unused by the linker
                     if target_triple.contains("-apple-") {
                         command.arg("-Clink-arg=-Wl,-u,_AUDITABLE_VERSION_INFO");
+                    } else if target_triple.ends_with("-msvc") {
+                        command.arg("-Clink-arg=/INCLUDE:AUDITABLE_VERSION_INFO");
                     } else {
                         command.arg("-Clink-arg=-Wl,--undefined=AUDITABLE_VERSION_INFO");
                     }

